### DISCUSSION TITLE
Selector widget lov problem

### DISCUSSION THREAD

**Discussion by ziganow** [Avaiga/taipy#259] [Q&A] [answered]
Hi,

I am wondering why selector lov={my_list} is not working. I get always a concatenated list items string displayed. For me its only possible to pass values as ";".join(my_list)

any suggestions

Thanks in advance

Alex

**Comment by FlorianJacta** [comment]
Here is a little code showing how to use a selector:

```python
from taipy.gui import Gui

list_of_elements = ['First element', 'Second element', 'Third element']
selected_value = None

md = """
<|{selected_value}|selector|lov={list_of_elements}|>

<|{selected_value}|>
"""

Gui(md).run()
```

Do you get the appropriate result?

![image](https://github.com/Avaiga/taipy/assets/98709993/e385fc26-2355-4d0e-95bb-2f672843bc16)

**Comment by ziganow** [comment]
Hi,

Thank you for that fast reply.

I am doing the same things ( take a look at screenshots)


[image001.png]
[image002.png]
[image003.png]

#######

I have also the same behavior with disabling the menu items. I can only pass the string with concatenated values like „menu1;menu2“

[image001.png]

It have tried a list, a tuple a dictionary. But only the concatenated string works for me
(Using tapir Lib V 2.4 and python 3.11.2)

Thanks in advance

A. Ziganow
___________________________________________________________

[image002.png]

SimplyTest GmbH
Geschäftsführer: Alexander Heimlich, Waldemar Siebert

Sitz der Gesellschaft: Nürnberg
Registergericht: Amtsgericht Nürnberg
Registernummer: HRB 35089
Telefon: +49 (0)911 373 96 700
www.simplytest.de

[image004.png][image006.png]

Am 02.08.2023 um 16:53 schrieb FlorianJacta ***@***.***>:



Here is a little code showing how to use a selector:

from taipy.gui import Gui

list_of_elements = ['First element', 'Second element', 'Third element']
selected_value = None

md = """
<|{selected_value}|selector|lov={list_of_elements}|>

<|{selected_value}|>
"""

Gui(md).run()

Do you get the appropriate result?

[image]<https://user-images.githubusercontent.com/98709993/257853180-e385fc26-2355-4d0e-95bb-2f672843bc16.png>

**Reply by ziganow** [reply]
Sorry if images are not visible
and sorry for typos (Using taipy Lib V 2.4 and python 3.11.2)
I have tried a list, a tuple a dictionary. But only the concatenated string works for me

![image001](https://github.com/Avaiga/taipy/assets/13311546/43a7ef0a-a57a-439e-938c-cbfe37eded92)

![image002](https://github.com/Avaiga/taipy/assets/13311546/2ce40169-f39f-4fe2-a0b1-75364b66b74c)

![image003](https://github.com/Avaiga/taipy/assets/13311546/72b1898d-1554-4bd5-bed2-532956f1b3ac)

![image004](https://github.com/Avaiga/taipy/assets/13311546/0914685d-e909-4228-9380-78cbf2ee114c)

**Comment by FlorianJacta** [comment, marked as answer]
Try to remove the `f` before the string. See my example above, there are no `f` before the string.

`f"""` -> `"""`

It should fix your issue.

**Comment by ziganow** [comment]
Thx,

that fixed the problem
